### PR TITLE
SONAR-7722 The project name should not be mandatory

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: sonar-plugin-api/src/main/java/org/sonar/api/batch/bootstrap/ProjectDefinition.java [hunk 1]
```diff
@@ -183,12 +183,9 @@ public String getVersion() {
...
   }
 
+  @CheckForNull
   public String getName() {
-    String name = properties.get(CoreProperties.PROJECT_NAME_PROPERTY);
-    if (StringUtils.isBlank(name)) {
-      name = "Unnamed - " + getKey();
-    }
-    return name;
+    return properties.get(CoreProperties.PROJECT_NAME_PROPERTY);
```
Comment (henryju) [inline sonar-plugin-api/src/main/java/org/sonar/api/batch/bootstrap/ProjectDefinition.java]: This is somehow breaking backward compatibility. What about logs that were logging each module name? Can't we keep a "non null" behavior, and keep a way to know the original value to populate the scanner report?

Comment (dbmeneses) [inline sonar-plugin-api/src/main/java/org/sonar/api/batch/bootstrap/ProjectDefinition.java]: Good catch, I added dedicated methods in Project and ProjectDefinition.

**Comment by henryju** [comment]
LGTM

File: sonar-scanner-engine/src/main/java/org/sonar/scanner/report/ComponentsPublisher.java [hunk 2]
```diff
@@ -149,6 +150,10 @@ private static String getLanguageKey(Resource r) {
 
   @CheckForNull
   private static String getName(Resource r) {
+    if (ResourceUtils.isProject(r)) {
```
Comment (SonarTech) [inline sonar-scanner-engine/src/main/java/org/sonar/scanner/report/ComponentsPublisher.java]: MINOR Remove this use of "ResourceUtils"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: sonar-scanner-engine/src/main/java/org/sonar/scanner/report/ComponentsPublisher.java [hunk 3]
```diff
@@ -149,6 +150,10 @@ private static String getLanguageKey(Resource r) {
...
+      Project project = (Project) r;
```
Comment (SonarTech) [inline sonar-scanner-engine/src/main/java/org/sonar/scanner/report/ComponentsPublisher.java]: MINOR Remove this use of "Project"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
MINOR Remove this use of "Project"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* MINOR 3 minor

Watch the comments in this conversation to review them.

**Pull request closed without merging**